src/general/base_stage/EditorComponentWithActionsBase.cs

```diff
@@ -135,11 +146,17 @@ protected virtual void OnCancelActionClicked()
     protected void Undo()
     {
         Editor.Undo();
+
+        // Update the finish badge warning visibility in case undoing an action caused or fixed the warning
+        UpdateFinishButtonWarningVisibility();
```

Comment:
Is this the right place to put this? As EditorBase has `[RunOnKeyDown("e_undo")]` it can be triggered from the keyboard without this being called. I think instead you'll want to hook into the editor "on action changed". So that'd be in the component's `UpdateUndoRedoButtons` method (though the name is a bit too specific to really warrant this, but I'll accept it by thinking about the warning badge being basically kind of a thing that is related to the currently performed action history).